### DISCUSSION TITLE
Allow the home button to clear search results

### DISCUSSION THREAD

**Discussion by mamekoro** [go-shiori/shiori#871] [Feature requests]
I'd like an easy way to return to the home screen after a search.
Could the search results be cleared when the home button is pressed?

**Comment by Monirzadeh** [comment]
currently if home is active you can't press home button.
it be weird if you clear search with switching between pages.
add a clear search field is not a better idea? that `X` icon is not the best ui i thinking about that but any better solution?
maybe make home page clickable all the time.
without search input
![without-search-input](https://github.com/go-shiori/shiori/assets/25131576/ef01ce4b-96a4-40b5-b2c3-868e3866b124)
when searchbox has text
![search-clear](https://github.com/go-shiori/shiori/assets/25131576/24655ded-119a-4f0f-bb24-0ba6f68180f3)
that `x` show just when field has input.

**Reply by mamekoro** [reply]
I checked Google, Bing and DuckDuckGo. They have the following UI at the top of their search results pages.

```
[Logo and/or brand name] [Search bar   [X button] [Search button]]
```

- `[Logo and/or brand name]` is clickable. When it is clicked, it discards the search results and opens a fresh home page. (The word "fresh" here means that no text is entered in the search bar.)
- `[X button]` is clickable. When it is clicked, it clears the search bar, but does not open the home page (i.e. the search results are retained).

In my opinion, we can have both the home button and the X button.

Both of the buttons should clear the text entered in the search bar.
Additionally, the home button should open a fresh home page, while the X button should retain existing search results.
